src/index.js

```diff
@@ -24,17 +36,180 @@ function openTab(i) {
+//         clearTabs();
+//         openTab(index);
+//     });
+// }
+
+function generateText(){
+    // Generating typing content
+    const paragraph = _paragraph(50);
+    content.innerHTML = paragraph;
+    // console.log(paragraph);
+
+    // To convert all the charecters to lower case
+    return smash = content.innerText;
+}
+
+function countDown(){
+    console.log('countdown started');
+    let haha = setInterval(() => {
+        // this has a delay of 1s, to compensate it time is shown as 'time-1' and only timeValue-1 intervals are requested.
+        if(time>1){
+            timeCount.innerHTML = (time-1);
+            time--;
+        }
+        else{
+            speed = w*2;
+            accuracy = Math.floor((i/k)*100);
+
+            timeCount.style.fontSize = '4rem';
+            timeCount.innerHTML = 'Typing Speed: '+ speed + 'WPM' + '</br>Accuracy: ' + accuracy + '%';
+
+            typeUpdate.disabled= true;
+            startButton.style.opacity = '1';
+            startButton.style.pointerEvents = 'auto';
+
+            clearInterval(haha);
+
+            // typeUpdate.removeEventListener('mousemove', validation);
+
+            time= timeValue;
+            startButton.disabled = false;
+            console.log(typeUpdate);
+        }
+    }, 1000);
+}
+
+function resetVariables(){
+    i= 0, j= 0, k= 0, w = 0, speed = 0, accuracy = 0;
+}
+
+// Verification of the entries (Did not work when outside of the event listner).
+// let validation = (e, f) => {
+//     let a = f[i];
+//     let b = e.key;
+//     console.log(a, b, i, "errors = " + j, k, w);
+//     k++;
+//     if(a==b){
+//         if(b == ' '){
+//             w++;
+//         };
+//         i++;
+//     }
+//     else if(b == 'Backspace'){
+//         if(b == ' '){
+//             w--;
+//         }
+//         i--;
+//     }
+//     else if(b == 'Shift' || b == 'Tab' || b == 'Alt'){
+//         console.log('hello you pressed an odd key');
+//     }
+//     else if (a != b){
+//         i++;
+//         j++;
+//     }
+// };
+
+window.onload = () => {
+    restartButton.style.display = 'none';
+    typeUpdate.disabled= true;
+
+    // generateText();
+
+    // to clear all the tabs when the page is loaded
+    clearTabs();
+
+    // to open the default tab wihch is tab[0]
+    openTab(0);
+}
+
+startButton.addEventListener('click', () => {
+    // console.log('test started');
+    startButton.style.opacity = '.5';
+    startButton.style.pointerEvents = 'none';
+
+    // console.log('yo this is the thing'+ smash);
+    resetVariables();
+    // console.log(i);
+
+    generateText();
+
+    typeUpdate.value = '';
+    typeUpdate.disabled= false;
+    typeUpdate.focus();
+    timeCount.style.fontSize = '7rem';
+    timeCount.innerHTML = timeValue;
+    countDown();
+
+    // validation
+    typeUpdate.addEventListener('keyup', (e) => {
+    let a = smash[i];
+    let b = e.key;
+    console.log(a, b, i, "errors = " + j, k, w);
+    k++;
+    if(a==b){
```

Comment:
fix: Prefer `===` over `==` . == is not recommended in javascript since it comes with some pains of type coercion
Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness